massa-api/src/private.rs

```diff
@@ -135,36 +147,64 @@ impl MassaRpcServer for API<Private> {
-        let network_command_sender = self.0.network_command_sender.clone();
-        network_command_sender
-            .node_ban_by_ips(ips)
-            .await
-            .map_err(|e| ApiError::NetworkError(e).into())
+    async fn node_ban_by_ip(&self, _ips: Vec<IpAddr>) -> RpcResult<()> {
+        // let network_command_sender = self.0.network_command_sender.clone();
+        // network_command_sender
+        //    .node_ban_by_ips(ips)
+        //    .await
+        //    .map_err(|e| ApiError::NetworkError(e).into())
+        return Err(
+            ApiError::BadRequest("This request is currently not available".to_string()).into(),
+        );
     }
 
     async fn node_ban_by_id(&self, ids: Vec<NodeId>) -> RpcResult<()> {
-        let network_command_sender = self.0.network_command_sender.clone();
-        network_command_sender
-            .node_ban_by_ids(ids)
-            .await
-            .map_err(|e| ApiError::NetworkError(e).into())
+        let protocol_controller = self.0.protocol_controller.clone();
+        //TODO: Change when unify node id and peer id
+        let peer_ids = ids
+            .into_iter()
+            .map(|id| {
+                PeerId::from_bytes(
+                    id.get_public_key().to_bytes()[..PUBLIC_KEY_SIZE_BYTES]
+                        .try_into()
+                        .unwrap(),
+                )
+                .unwrap()
+            })
+            .collect();
+        protocol_controller
+            .ban_peers(peer_ids)
+            .map_err(|e| ApiError::ProtocolError(e).into())
     }
 
     async fn node_unban_by_id(&self, ids: Vec<NodeId>) -> RpcResult<()> {
-        let network_command_sender = self.0.network_command_sender.clone();
-        network_command_sender
-            .node_unban_by_ids(ids)
-            .await
-            .map_err(|e| ApiError::NetworkError(e).into())
+        let protocol_controller = self.0.protocol_controller.clone();
+        //TODO: Change when unify node id and peer id
+        let peer_ids = ids
+            .into_iter()
+            .map(|id| {
+                PeerId::from_bytes(
+                    id.get_public_key().to_bytes()[..PUBLIC_KEY_SIZE_BYTES]
+                        .try_into()
+                        .unwrap(),
+                )
+                .unwrap()
+            })
+            .collect();
+        protocol_controller
+            .unban_peers(peer_ids)
+            .map_err(|e| ApiError::ProtocolError(e).into())
     }
 
-    async fn node_unban_by_ip(&self, ips: Vec<IpAddr>) -> RpcResult<()> {
-        let network_command_sender = self.0.network_command_sender.clone();
-        network_command_sender
-            .node_unban_ips(ips)
-            .await
-            .map_err(|e| ApiError::NetworkError(e).into())
+    async fn node_unban_by_ip(&self, _ips: Vec<IpAddr>) -> RpcResult<()> {
+        // let network_command_sender = self.0.network_command_sender.clone();
```

Comment:
todo here ?

Reply:
Ok. Added. Anyway I will open issue for the API endpoint that will be temporary disabled when this PR is merged